Signal/src/Profiles/OWSProfileManager.m

```diff
@@ -623,11 +637,154 @@ - (void)fetchProfileForRecipientId:(NSString *)recipientId
     }
 
     userProfile.lastUpdateDate = [NSDate new];
-    [self.dbConnection readWriteWithBlock:^(YapDatabaseReadWriteTransaction *transaction) {
-        [userProfile saveWithTransaction:transaction];
-    }];
 
-    // TODO: Actually update the profile.
+    [self saveUserProfile:userProfile];
+
+    [ProfileFetcherJob runWithRecipientId:recipientId
+                           networkManager:self.networkManager
+                         ignoreThrottling:ignoreThrottling];
```

Comment:
> It's a little weird to have one timeout clobber another.

First of all, this `ignoreThrottling` might be excessive.  The intention was to ensure that we refetched whenever a profile key changed, but since profile key changes (i.e. re-registration) should be rare, this shouldn't be a big deal.  It'll help with testing and be a slightly more crisp experience when a contact re-registers.

> An interesting consequence of this thought exercise is I realize we'll now that scrolling through the inbox triggers profile fetches, and triggers any learned SN changes.

That's absolutely right - there was now _way more_ triggers for profile fetches.  This will work in the "new conversation" views too, which is nice.  

>  It's a little weird to have one timeout clobber another.

I'm not sure what you mean - it's not two separate timeouts, it's an event (profile key change) that triggers an immediate, unthrottled fetch.